### ISSUE TITLE
error when trying `get_destructions` API request

### ISSUE THREAD

**Issue by jdogresorg** [CounterpartyXCP/counterparty-lib#1195]
when requesting `get_destructions` for block 747248 is resulting in an error being thrown.

Request
---
```
{
    "method": "get_destructions",
    "params": {
        "filters": [],
        "order_by": "tx_index",
        "order_dir": "ASC",
        "start_block": 747248,
        "end_block": 747248
    },
    "jsonrpc": "2.0",
    "id": 0
}
```

Response
---
```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>500 Internal Server Error</title>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or
	there is an error in the application.</p>
```

After some digging it appears that the only destruction that happened in the problem block is this one https://xchain.io/tx/2074497

And the `tag` field in the destruction has a funky character in it which is most likely causing the issue

**Comment by jdogresorg** [comment]
here is the error thrown from the CP Console

```
[2022-08-12 16:49:19][ERROR] API Exception: {'type': 'UnicodeDecodeError', 'args': ('utf-8', b'NORMANCOMICS FAKE RARE OFFICIAL N\xc2', 33, 34, 'unexpected end of data'), 'message': "'utf-8' codec can't decode byte 0xc2 in position 33: unexpected end of data"}
counterparty_1           | Traceback (most recent call last):
counterparty_1           |   File "/usr/local/lib/python3.6/dist-packages/jsonrpc/manager.py", line 108, in _get_responses
counterparty_1           |     result = method(*request.args, **request.kwargs)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/api.py", line 518, in get_method
counterparty_1           |     return get_rows(self.db, table=table, **kwargs)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/api.py", line 282, in get_rows
counterparty_1           |     return adjust_get_destructions_results(query_result)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/api.py", line 294, in adjust_get_destructions_results
counterparty_1           |     destruction_row['tag'] = destruction_row['tag'].decode('utf-8')
counterparty_1           | UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc2 in position 33: unexpected end of data
counterparty_1           | [2022-08-12 16:49:19,872] ERROR in app: Exception on / [POST]
counterparty_1           | Traceback (most recent call last):
counterparty_1           |   File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1988, in wsgi_app
counterparty_1           |     response = self.full_dispatch_request()
counterparty_1           |   File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1641, in full_dispatch_request
counterparty_1           |     rv = self.handle_user_exception(e)
counterparty_1           |   File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1544, in handle_user_exception
counterparty_1           |     reraise(exc_type, exc_value, tb)
counterparty_1           |   File "/usr/local/lib/python3.6/dist-packages/flask/_compat.py", line 33, in reraise
counterparty_1           |     raise value
counterparty_1           |   File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1639, in full_dispatch_request
counterparty_1           |     rv = self.dispatch_request()
counterparty_1           |   File "/usr/local/lib/python3.6/dist-packages/flask/app.py", line 1625, in dispatch_request
counterparty_1           |     return self.view_functions[rule.endpoint](**req.view_args)
counterparty_1           |   File "/usr/local/lib/python3.6/dist-packages/flask_httpauth.py", line 88, in decorated
counterparty_1           |     return f(*args, **kwargs)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/api.py", line 955, in handle_root
counterparty_1           |     response = handle_rpc_post(request_json)
counterparty_1           |   File "/counterparty-lib/counterpartylib/lib/api.py", line 1008, in handle_rpc_post
counterparty_1           |     response = flask.Response(jsonrpc_response.json.encode(), 200, mimetype='application/json')
counterparty_1           |   File "/usr/local/lib/python3.6/dist-packages/jsonrpc/base.py", line 85, in json
counterparty_1           |     return self.serialize(self.data)
counterparty_1           |   File "/usr/lib/python3.6/json/__init__.py", line 231, in dumps
counterparty_1           |     return _default_encoder.encode(obj)
counterparty_1           |   File "/usr/lib/python3.6/json/encoder.py", line 199, in encode
counterparty_1           |     chunks = self.iterencode(o, _one_shot=True)
counterparty_1           |   File "/usr/lib/python3.6/json/encoder.py", line 257, in iterencode
counterparty_1           |     return _iterencode(o, 0)
counterparty_1           |   File "/usr/lib/python3.6/json/encoder.py", line 180, in default
counterparty_1           |     o.__class__.__name__)
counterparty_1           | TypeError: Object of type 'bytes' is not JSON serializable
```

**Comment by jotapea** [comment]
Thank you @jdogresorg for creating this. Yes I can confirm the issue is when generating a response that includes the block 747248.

A request like the following fails:

```
{
  "jsonrpc": "2.0",
  "id": 0,
  "method": "get_destructions",
  "params": {
    "filters": [],
    "order_by": "tx_index",
    "order_dir": "ASC",
    "end_block": 747248,
    "offset": 4000
  }
}
```

But without the offset, which won't include block 747248 in the response, works.

**Comment by jdogresorg** [comment]
this should be fixed with PR #1198
